app/components/HeaderNotifications/index.js

```diff
@@ -80,11 +61,7 @@ export default class NotificationsDropdown extends Component<Props, State> {
     return (
       <div>
         {notifications.map((notification, i) => (
-          <NotificationElement
-            key={i}
-            notification={notification}
-            markNotification={this.markNotification}
-          />
+          <NotificationElement key={i} notification={notification} />
```

Comment:
Using the index of an array as key has turned out to be dangerous. If an element is deleted, the elements are moved around so that a new element has the same key as the deleted one, which can lead to weird behaviour. Might not be an issue here as I don't know if notifications can be deleted, but it seems like a good plan to use something like `key={notifiaction.id}` instead just in case.

Reply:
Ok, I'll fix that

Reply:
It is not necessarily better @finnss since the notification id might not be unique either. A better example would be something like
``` key={`notification-${i}`} ``` which would create unique keys like `notification-1` etc. (As long as nobody else creates the same string as key somewhere else)

Reply:
isn't the notification id from the database? o.o
and why is `notification-${i}` any different from just `i`? (you still have the same bug with sorting/deleting etc.)

pretty sure the id needs to be something that is unique only for the siblings of that dom element, not globally in the app

Reply:
@hanswilw & @ekmartin 

Using the index from mapping/looping as the react-key is an anti pattern that creates strange bugs, that are hard to identify. Since it we never "remove" one notification it isn't "that" important, but when new notifications are added, using indexes will result in way more unnecessary rendering - aka. it only removes the warning. It will then create new elements in the bottom of the list, and moves the old elements further down.

Keys only have to be unique among their siblings, and they _don't_ need to be unique globally. 

I used quite a bit of time to track down #848, that was a result of using mapping indexes as keys. When deleting an element in the middle, it is the react-component in the bottom that is unmounted and removed, and the state-elements are popped upwards to fill the gap.

This is generally not a huge problem with other than performance when using stateless components, but when state is added to components - this __will__ create bugs.

Reply:
I bow, I was told otherwise at some point earlier – thats why I commented. Should've checked it out better myself though 😌